universum/modules/vcs/github_actions_vcs.py

```diff
@@ -0,0 +1,119 @@
+import json
+import urllib.parse
+
+from . import git_vcs
+from ..reporter import ReportObserver, Reporter
+from ...lib import utils
+from ...lib.gravity import Dependency
+
+__all__ = [
+    "GithubActionsMainVcs"
+]
+
+
+class GithubActionsMainVcs(ReportObserver, git_vcs.GitMainVcs):
+    """
+    This class mostly contains functions for Gihub report observer
+    """
+    reporter_factory = Dependency(Reporter)
+
+    @staticmethod
+    def define_arguments(argument_parser):
+        parser = argument_parser.get_or_create_group("GitHub Actions", "GitHub repository settings for GH Actions")
+
+        parser.add_argument("--ghactions-token", "-ght", dest="token", metavar="GITHUB_TOKEN",
+                            help="Is stored in ${{ secrets.GITHUB_TOKEN }}")
+        parser.add_argument("--ghactions-payload", "-ghp", dest="payload", metavar="GITHUB_PAYLOAD",
+                            help="File path: ${{ github.event_path }}")
+
+    def __init__(self, *args, **kwargs):
+        self.settings.repo = "Will be filled after payload parsing"
+        super().__init__(*args, **kwargs)
+        self.reporter = None
+
+        self.check_required_option("token", """
+            The GitHub workflow token is not specified.
+
+            For github the git checkout id defines the commit to be checked and reported.
+            Please specify the checkout id by using '--git-checkout-id' ('-gco') command
+            line parameter or by setting GIT_CHECKOUT_ID environment variable.
+
+            If using 'universum github-handler', the checkout ID is automatically extracted
+            from the webhook payload and passed via GIT_CHECKOUT_ID environment variable.
+            """)
+
+        self.payload = self.read_and_check_multiline_option("payload", """
+            GitHub web-hook payload JSON is not specified.
+
+            Please pass incoming web-hook request payload to this parameter directly via
+            '--github-payload' ('-ghp') command line parameter or by setting GITHUB_PAYLOAD
+            environment variable, or by passing file path as the argument value (start
+            filename with '@' character, e.g. '@/tmp/file.json' or '@payload.json' for
+            relative path starting at current directory). Please note, that when passing
+            a file, it's expected to be in UTF-8 encoding
+            """)
+
+        try:
+            self.payload_json = json.loads(self.payload)
+            self.settings.repo = self.payload_json['repository']['html_url']
+            self.settings.refspec = self.payload_json['pull_request']['head']['ref']
+        except json.decoder.JSONDecodeError as error:
```

Comment:
Hande the `KeyError` also, as `self.payload` contains user input.
There are same issues with user-provided JSON usage without any checks below. Use the `.get()` method or exception handling depending on the situation.